src/java/org/apache/cassandra/exceptions/RequestFailureReason.java

```diff
@@ -81,25 +92,18 @@ public static RequestFailureReason fromCode(int code)
             throw new IllegalArgumentException("RequestFailureReason code must be non-negative (got " + code + ')');
 
         // be forgiving and return UNKNOWN if we aren't aware of the code - for forward compatibility
-        return code < codeToReasonMap.length ? codeToReasonMap[code] : UNKNOWN;
+        return codeToReasonMap.getOrDefault(code, UNKNOWN);
     }
 
     public static RequestFailureReason forException(Throwable t)
     {
-        if (t instanceof TombstoneOverwhelmingException)
-            return READ_TOO_MANY_TOMBSTONES;
-
-        if (t instanceof IncompatibleSchemaException)
-            return INCOMPATIBLE_SCHEMA;
-
-        if (t instanceof NotCMSException)
-            return NOT_CMS;
-
-        if (t instanceof InvalidRoutingException)
-            return INVALID_ROUTING;
+        RequestFailureReason r = exceptionToReasonMap.get(t.getClass());
+        if (r != null)
+            return r;
 
-        if (t instanceof CoordinatorBehindException)
-            return COORDINATOR_BEHIND;
+        for (Map.Entry<Class<? extends Throwable>, RequestFailureReason> entry : exceptionToReasonMap.entrySet())
+            if (entry.getKey().isInstance(t))
+                return entry.getValue();
```

Comment:
So `forException()` is actually now mapping `TIMEOUT`, `READ_SIZE`, and `INDEX_NOT_AVAILABLE` where it wasn't before, correct? Just wanted to make sure that was intended (and a little curious about why there weren't already mapped...)

Reply:
Those were missed. So, I am adding the mapping and the brittle way of checking the length now to make sure people deliberately add or decide not to add them here. The drivers use this method with v5, where the `reasonMap` exists. It occurs to me that people just missed adding those; it's not hard to do.